plugins/lua/tiletypes.lua

```diff
@@ -0,0 +1,29 @@
+local _ENV = mkmodule('plugins.tiletypes')
+
+local utils = require('utils')
+
+local function parse_cursor(opts, arg)
```

Comment:
Can this be shared with `plugins/lua/blueprint.lua` somehow? Seems that the primary difference is the field name, but the function could likely be refactored to return a `df.coord`.

Reply:
yeah, let me put a common implementation in `utils.parseCoords()`. well, actually it might be better to use the implementation from `dig-now`, which supports "here" as a coordinate. That pulls in guidm, though, for the cursor position, which would be a circular dependency. Perhaps a new file for common argument parsing functions? `library/lua/argparse.lua`? And we can migrate the `processArgs*` functions there as well.

Reply:
We probably want to keep `processArgs*` in utils.lua (at least as stubs) for compatibility, so I'm not sure how best to resolve the circular import. What about lazily requiring `guidm` for the cursor position check (`require('gui.dwarfmode').getCursorPos()`)?

Reply:
(Open to also moving argument parsing to `argparse.lua` as long as scripts using `processArgs*` don't need to be updated)

Reply:
done. this PR now depends on #1897